src/System.Net.Requests/tests/RequestStreamTest.cs

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Xunit;
+
+namespace System.Net.Tests
+{
+    public class RequestStreamTest
+    {
+        readonly byte[] buffer = new byte[1];
+        
+        [Fact]
+        public void WriteAsync_BufferIsNull_ThrowsArgumentNullException()
+        {
+            using (Stream stream = GetRequestStream())
+            {
+                Assert.Throws<ArgumentNullException>(() => { Task t = stream.WriteAsync(null, 0, 1); });
+            }
+        }
+
+        [Fact]
+        public void WriteAsync_OffsetIsNegative_ThrowsArgumentOutOfRangeException()
+        {
+            using (Stream stream = GetRequestStream())
+            {
+                Assert.Throws<ArgumentOutOfRangeException>(() => { Task t = stream.WriteAsync(buffer, -1, buffer.Length); });
+            }
+        }
+
+        [Fact]
+        public void WriteAsync_CountIsNegative_ThrowsArgumentOutOfRangeException()
+        {
+            using (Stream stream = GetRequestStream())
+            {
+                Assert.Throws<ArgumentOutOfRangeException>(() => { Task t = stream.WriteAsync(buffer, 0, -1); });
+            }
+        }
+
+        [Fact]
+        public void WriteAsync_OffsetPlusCountExceedsBufferLength_ThrowsArgumentException()
+        {
+            using (Stream stream = GetRequestStream())
+            {
+                Assert.Throws<ArgumentException>(() => { Task t = stream.WriteAsync(buffer, 0, buffer.Length+1); });
+            }
+        }
+
+        [Fact]
+        public void WriteAsync_OffsetPlusCountMaxValueExceedsBufferLength_ThrowsArgumentException()
+        {
+            using (Stream stream = GetRequestStream())
+            {
+                Assert.Throws<ArgumentException>(() => { Task t = stream.WriteAsync(buffer, int.MaxValue, int.MaxValue); });
+            }
+        }
+
+        [Fact]
+        public void WriteAsync_ValidParameters_Success()
+        {
+            using (Stream stream = GetRequestStream())
+            {
+                stream.WriteAsync(buffer, 0, buffer.Length).GetAwaiter().GetResult();
+            }
+        }
+
+        [Fact]
+        public void WriteAsync_TokenIsCanceled_TaskIsCanceled()
+        {
+            using (Stream stream = GetRequestStream())
+            {
+                var cts = new CancellationTokenSource();
+                cts.Cancel();
+                Task t = stream.WriteAsync(buffer, 0, buffer.Length, cts.Token);
+                Assert.True(t.IsCanceled);
+            }
+        }
+
+        [Fact]
+        public void FlushAsync_Success()
+        {
+            using (Stream stream = GetRequestStream())
+            {
+                stream.FlushAsync().GetAwaiter().GetResult();
```

Comment:
This should also probably verify the Task.Status rather than using GetResult like this. With the current code, you won't know if the returned task actually completed asynchronously.